### PR TITLE
Variable rendering improvements

### PR DESCRIPTION
**Pull request by rotemavni**
The purpose of this PR is to make sure all the tests in `tests/terraform/parser/test_parser_scenarios.py` pass for the graph's variable rendering algorithm.
**During the testing some bugs were detected and fixed:**
1. Locals blocks were created with a config that includes all locals in the same block.
2. When evaluating edges during the variable rendering, if you have several edges in the same iteration, sort the edges in an ascending order, determined by the dest vertex out degree. This way, the more edges coming out of the dest vertex, we want to evaluate it later.
3. Several problems in updating the config after the rendering, involving the wrapping `[]` of the tf_definitions.
4. Added support of multiple terraform functions
5. Better handling of interpolations.
6. Evaluate every string attribute that wasn't rendered.

**Tests**
The test cases from `test_parser_scenarios` are ran now in `test_render_scenarios`.
Some of the test cases had to be slightly modified.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: checkov/graph/terraform/variable_rendering/evaluate_terraform.py [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
 # condition ? true_val : false_val -> (condition, true_val, false_val)
+from checkov.terraform.parser_utils import find_var_blocks
+
 CONDITIONAL_EXPR = r'([^\s]+)\?([^\s^\:]+)\:([^\s^\:]+)'
 
 # {key1 = value1, key2 = value2, ...}
 MAP_REGEX = r'\{(?:\s*[\S]+\s*\=\s*[\S]+\s*\,)+(?:\s*[\S]+\s*\=\s*[\S]+\s*)\}'
 
+# {key:val}[key]
+MAP_WITH_ACCESS = r'(?P<d>\{(?:\s*[\S]+\s*\:\s*[\S]+\s*)+(\,?:\s*[\S]+\s*\:\s*[\S]+\s*)*\})\s*(?P<access>\[\S+\])'
```
Comment (nimrodkor) [inline checkov/graph/terraform/variable_rendering/evaluate_terraform.py:15]: @rotemavni Love the group name! `<access>`

File: checkov/graph/terraform/variable_rendering/evaluate_terraform.py [hunk 2]
```diff
@@ -41,17 +50,33 @@ def _try_evaluate(input_str):
...
+    # get all variable references in string
+    # remove from the string all ${} or '${}' occurrences
+    var_blocks = find_var_blocks(original_str)
+    var_blocks.reverse()
+    for block in var_blocks:
+        if block.full_str.startswith("${") and block.full_str.endswith("}"):
+            full_str_start = original_str.find(block.full_str)
+            full_str_end = full_str_start + len(block.full_str)
+            if full_str_start > 0 and full_str_end < len(original_str) - 2 and original_str[full_str_start-1] == "'" and original_str[full_str_start-1] == original_str[full_str_end] and "." in block.full_str:
+                # checking if ${} is wrapped with '' like : '${}'
```
Comment (nimrodkor) [inline checkov/graph/terraform/variable_rendering/evaluate_terraform.py:76]: @rotemavni I believe terraform supports `"`, not `'` right?

File: checkov/graph/terraform/variable_rendering/renderer.py [hunk 3]
```diff
@@ -4,14 +4,16 @@
 
 from checkov.graph.terraform.graph_builder.graph_components.attribute_names import CustomAttributes
 from checkov.graph.terraform.graph_builder.graph_components.block_types import BlockType
-from checkov.graph.terraform.utils.utils import get_referenced_vertices_in_value, run_function_multithreaded, calculate_hash, \
-    join_trimmed_strings, remove_interpolation, remove_index_pattern_from_str, remove_function_calls_from_str
+from checkov.graph.terraform.utils.utils import get_referenced_vertices_in_value, run_function_multithreaded, \
+    calculate_hash, \
+    join_trimmed_strings, remove_index_pattern_from_str
```
Comment (nimrodkor) [inline checkov/graph/terraform/variable_rendering/renderer.py:9]: Can be oneline

**Review by nimrodkor — COMMENTED** [review]
@rotemavni Nice!
Can we comment out the tests that were migrated?

File: checkov/graph/terraform/variable_rendering/evaluate_terraform.py (hunk 2 above)
Comment (rotemavni) [reply checkov/graph/terraform/variable_rendering/evaluate_terraform.py:76]: @nimrodkor It does, but in the evaluation process python puts `'`  and we're trying to eliminate them

**Comment by rotemavni** [comment]
> @rotemavni Nice!
> Can we comment out the tests that were migrated?

I made some changes in the tests and as long as we don't deprecate the other parsing I feel more comfortable with keeping these tests

**Review by nimrodkor — APPROVED** [review]

**Pull request merged**